Missing cast causes build to fail

Hey 👋🏼 

we are using `flint` as a dependency in one of our projects (https://github.com/cda-tum/qecc) and have to build it from source as part of the Python wheel building process. A week ago, our builds suddenly started failing (see, e.g., https://github.com/cda-tum/qecc/actions/runs/4114252110/jobs/7101380386). The error message is
> /usr/local/include/flint/nmod_mat.h: In function ‘void nmod_mat_permute_rows(nmod_mat_struct*, const mp_limb_signed_t*, mp_limb_signed_t*)’:
  /usr/local/include/flint/nmod_mat.h:365:40: error: invalid conversion from ‘void*’ to ‘mp_limb_t**’ {aka ‘long unsigned int**’} [-fpermissive]
    365 |     mp_limb_t ** mat_tmp = flint_malloc(mat->r * sizeof(mp_limb_t *));

The corresponding function was added in https://github.com/flintlib/flint2/pull/1208 a couple weeks ago.
While I am not at all familiar with your code base, I'd guess this is simply missing a proper cast and should be
```c++
mp_limb_t ** mat_tmp = (mp_limb_t **) flint_malloc(mat->r * sizeof(mp_limb_t *));
```
here
https://github.com/flintlib/flint2/blob/46711eda00f434b491cee3e8b8b225986889705b/nmod_mat.h#L365

--- a/nmod_mat.h
+++ b/nmod_mat.h
@@ -362,7 +362,7 @@ nmod_mat_permute_rows(nmod_mat_t mat,
                       slong * perm_store)
 {
 		slong i;
-    mp_limb_t ** mat_tmp = flint_malloc(mat->r * sizeof(mp_limb_t *));
+    mp_limb_t ** mat_tmp = (mp_limb_t **) flint_malloc(mat->r * sizeof(mp_limb_t *));
 
     /* perm_store[i] <- perm_store[perm_act[i]] */
     if (perm_store)

